### PR TITLE
Initial logging impl

### PR DESCRIPTION
**Pull request by guygrigsby**
First pass at a logging impl

### CODE REVIEW

File: test/log_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package test
+
+import (
+	"github.com/go-kit/log/level"
+	"github.com/livecontrol-io/logger"
+)
+
+func ExampleLevel() {
```
Comment (mkrufky) [inline test/log_test.go]: Can we pass `t *testing.T` in as an argument, then `t.Error()` if the resulting string doesn't match what we expect?

To do that, you'll probably have to pass an `io.Writer` or something else into `kitlog.NewSyncWriter()` within your `New()` function.  The replacement `io.Writer` can use `string.Builder` to build a string that can be compared within the test.

...unless there's some other way to mock out a logger without rewriting the entire package.

**Review by mkrufky — COMMENTED** [review]
Looks good 👍

File: test/log_test.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package test
```
[the remark this answers is not in the document]
Comment (guygrigsby) [reply test/log_test.go:1]: hmm. I actually disagree with that. We are testing that it complies to the interface

File: test/log_test.go (hunk 1 above)
Comment (guygrigsby) [reply test/log_test.go]: THis actually tests the output.
```
[~/g/s/g/l/l] [init] ➜  go test ./... -v
?   	github.com/livecontrol-io/logger	[no test files]
=== RUN   ExampleBasic
--- FAIL: ExampleBasic (0.00s)
got:
{"key":"val","level":"info","msg":"I like burritos"}
want:
{"key":"val","msg":"I like burritos"}
=== RUN   ExampleLevel
--- FAIL: ExampleLevel (0.00s)
got:
{"err":"an error","level":"error"}
want:
{"err":"an error"}
FAIL
FAIL	github.com/livecontrol-io/logger/test	0.132s
FAIL
```
source https://go.dev/blog/examples

File: test/log_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (guygrigsby) [reply test/log_test.go:1]: In the example for [testable examples in the std lib](https://go.dev/blog/examples), they use an external package `stringutil_test`. I would agree with you for unit tests testing individual functions, but this is a testable example for external use.

**Comment by guygrigsby** [comment]
Shit. This is a repo in my own account

File: test/log_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (halverneus) [reply test/log_test.go:1]: What do you all think about standardizing on an "example" folder for package usage? Either
A: /example/example.go (contents include various Example* functions)
B: /example/basic/example.go (contents include a single Example* function)

**Pull request merged**